exercises/fizzy/tests/fizzy.rs

```diff
@@ -0,0 +1,125 @@
+use fizzy::*;
+
+macro_rules! expect {
+    () => {
+        vec![
+            "1", "2", "fizz", "4", "buzz", "fizz", "7", "8", "fizz", "buzz", "11", "fizz", "13",
+            "14", "fizzbuzz", "16",
+        ]
+    };
+}
+
+#[test]
+fn test_simple() {
+    let got = fizz_buzz().apply(1..=16).collect::<Vec<_>>();
+    assert_eq!(expect!(), got);
+}
+
+#[test]
+#[ignore]
+fn test_u8() {
+    let got = fizz_buzz().apply(1_u8..=16).collect::<Vec<_>>();
+    assert_eq!(expect!(), got);
+}
+
+#[test]
+#[ignore]
+fn test_u64() {
+    let got = fizz_buzz().apply(1_u64..=16).collect::<Vec<_>>();
+    assert_eq!(expect!(), got);
+}
+
+#[test]
+#[ignore]
+fn test_nonsequential() {
+    let collatz_12 = &[12, 6, 3, 10, 5, 16, 8, 4, 2, 1];
+    let expect = vec![
+        "fizz", "fizz", "fizz", "buzz", "buzz", "16", "8", "4", "2", "1",
+    ];
+    let got = fizz_buzz()
+        .apply(collatz_12.into_iter().cloned())
+        .collect::<Vec<_>>();
+    assert_eq!(expect, got);
+}
+
+#[test]
+#[ignore]
+fn test_custom() {
+    let expect = vec![
+        "1", "2", "Fizz", "4", "Buzz", "Fizz", "Bam", "8", "Fizz", "Buzz", "11", "Fizz", "13",
+        "Bam", "BuzzFizz", "16",
+    ];
+    let fizzer = Fizzy::new()
+        .add_matcher(Matcher::new(|n| n % 5 == 0, "Buzz"))
+        .add_matcher(Matcher::new(|n| n % 3 == 0, "Fizz"))
+        .add_matcher(Matcher::new(|n| n % 7 == 0, "Bam"))
+        .apply(1..=16);
+    let got = fizzer.collect::<Vec<_>>();
+    assert_eq!(expect, got);
+}
+
+#[test]
+#[ignore]
+fn test_f64() {
+    // a tiny bit more complicated becuase range isn't natively implemented on floats
+    // NOTE: this test depends on a language feature introduced in Rust 1.34. If you
```

Comment:
These last two tests require [`std::iter::successors`](https://doc.rust-lang.org/std/iter/fn.successors.html), a 1.34 feature. It would be possible to work around it and support an earlier compiler, at the cost of a more complex test implementation in each case. 

I know of no track-wide policy on the minimum supported rustc version. We should decide between:

- it is ok for an exercise to require use of the most recent compiler
- it is ok for an exercise to require use of the most recent compiler if it includes an apologetic note (as here)
- the track as a whole can require no feature newer than some specified compiler version

Reply:
My preference, at strength 1/10, is to allow the most recent stable compiler. The reason for this preference is in the service of keeping the code expressive (high signal to noise ratio), and that I personally am able to stay sufficiently up to date.

Useful clarifying point about the note: When 1.34 is sufficiently old, would the note be eligible for removal?

If one goal of the track is to optimise for accessibility, we may be required to only use some version that is widely available. For example, if a significant portion of students use distribution D and that distribution's package manager contains Rust at 1.32, then perhaps we would choose to stay at 1.32.

I do not know of the characteristics of the students, so I can't make decisions on that dimension. We may have to wait and see.

Reply:
I agree: I also have no trouble keeping up with the most recent stable compiler, and therefore feel like it a reasonable thing to require of the students. We can revisit this if we start seeing student issues complaining that they don't have access to a recent stable compiler, but for now, let's decree the Official Track Policy: exercises may require use of the most recent compiler.